pkg/kubelet/kubelet_node_status.go

```diff
@@ -486,7 +486,7 @@ func (kl *Kubelet) setNodeAddress(node *v1.Node) error {
 			kl.cloudproviderRequestMux.Unlock()
 
 			go func() {
-				nodeAddresses, err = instances.NodeAddresses(context.TODO(), kl.nodeName)
+				kl.cloudproviderLastNodeAddresses, kl.cloudproviderLastError = instances.NodeAddresses(context.TODO(), kl.nodeName)
```

Comment:
i may be missing something obvious, but seems like cloudProviderLastError can just move next to declaration of var err error above on line 476

Reply:
The culprit is actually the `case <-kl.cloudproviderRequestSync` case. Each time the go routine asking for the node addresses finishes, it sends a value to the `kl.cloudproviderRequestSync` channel. If the happens after the `kl.cloudproviderRequestTimeout`, the `kl.cloudproviderRequestSync` is non-empty and in the next node status iteration the select picks the first case `<-kl.cloudproviderRequestSync` instead of waiting for the go routine to finish. Given the `nodeAddresses` and `err` are local variables and are both nil (cause the go routine responsible for setting them stores the values into previous local variables which no longer exist in the `setNodeAddress` scope), the `nodeAddresses` is always nil from the moment the timeout occurs.

This we need to store both (the list of node addresses and the error) into variables that are not local to the `setNodeAddress` method.

Reply:
there must be a simpler way to do this.  we've added 6 fields to the kubelet struct to support the timeout in this single area of the code.  i'm not saying i see it yet.  but there must be.  i'm too tired to see it atm though.

Reply:
this looks like it is writing to member variables asynchronously outside a lock... isn't that a crashing data race?

Reply:
Unlikely to happen but it is. Just the chance is very low. I am addressing this in another PR.

Reply:
Ok, I will update this PR with the suggested refactoring